mlflow/transformers.py

```diff
@@ -2149,6 +2189,71 @@ def _convert_cast_lists_from_np_back_to_list(data):
                     parsed_data.append(entry)
             return parsed_data
 
+    @staticmethod
+    def _convert_automatic_speech_recognition_input(data):
+        """
+        Conversion utility for decoding the base64 encoded bytes data of a raw soundfile when
+        parsed through model serving, if applicable. Direct usage of the pyfunc implementation
+        outside of model serving will treat this utility as a noop.
+
+        For reference, the expected encoding for input to Model Serving will be:
+
+        import requests
+        import base64
+
+        response = requests.get("https://www.my.sound/a/sound/file.wav")
+        encoded_audio = base64.b64encode(response.content).decode("ascii")
+
+        inference_data = json.dumps({"inputs": [encoded_audio]})
+
+        or
+
+        inference_df = pd.DataFrame(
+        pd.Series([encoded_audio], name="audio_file")
+        )
+        split_dict = {"dataframe_split": inference_df.to_dict(orient="split")}
+        split_json = json.dumps(split_dict)
+
+        or
+
+        records_dict = {"dataframe_records": inference_df.to_dict(orient="records")}
+        records_json = json.dumps(records_dict)
+
+        This utility will convert this JSON encoded, base64 encoded text back into bytes for
+        input into the AutomaticSpeechRecognitionPipeline for inference.
+        """
+
+        def is_base64(s):
+            try:
+                return base64.b64encode(base64.b64decode(s)) == s
+            except binascii.Error:
+                return False
+
+        def decode_audio(encoded):
+            if isinstance(encoded, bytes):
+                # For input types 'dataframe_split' and 'dataframe_records', the encoding
+                # conversion to bytes is handled.
+                if not is_base64(encoded):
+                    return encoded
```

Comment:
A bit confused here -  if `encoded` is of type `bytes`, can it every already be base64-encoded? I.e. isn't the output of base64 encoding always a string?

Reply:
Nevermind, it's always bytes. Neat!